frontend/src/metabase/search/components/filters/CreatedByFilter/CreatedByContent.tsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react/prop-types */
+import { useState } from "react";
+import { isEqual } from "underscore";
+import { t } from "ttag";
+import { useUserListQuery } from "metabase/common/hooks/use-user-list-query";
+import type { UserListResult } from "metabase-types/api";
+import { Center, Loader, Text, TextInput } from "metabase/ui";
+import {
+  CreatedByContainer,
+  CreatedByContentContainer,
+} from "metabase/search/components/filters/CreatedByFilter/CreatedByContent.styled";
+import type { SearchSidebarFilterComponent } from "metabase/search/types";
+import { UserListElement } from "metabase/search/components/filters/CreatedByFilter/UserListElement";
+
+export const CreatedByContent: SearchSidebarFilterComponent<"created_by">["ContentComponent"] =
+  ({ value, onChange }) => {
+    const { data: users = [], isLoading } = useUserListQuery();
+    const [userFilter, setUserFilter] = useState("");
+
+    const filteredUsers = users.filter(user =>
+      user.common_name.toLowerCase().includes(userFilter.toLowerCase()),
```

Comment:
The search algorithm should also consider the case where `user.common_name` is not present - we do [have a fallback in UI for it](https://github.com/metabase/metabase/blob/0540b7faec744708cb333e4a33a8405e02c00eb8/frontend/src/metabase/search/components/filters/CreatedByFilter/UserListElement.tsx#L25), so the search algorithm should match it.

Reply:
Pulled the logic into a util function, should be consistent across the two areas now